SU2_DEF/src/SU2_DEF.cpp

```diff
@@ -263,11 +263,34 @@ int main(int argc, char *argv[]) {
   /*--- Surface grid deformation using design variables ---*/
   
   for (iZone = 0; iZone < nZone; iZone++){
+
+    /*--- Initialize total deformation of design variables to zero ---*/
+    su2double **TotalDeformation;
+    unsigned short iDV, iDV_Value;
+    TotalDeformation = new su2double*[config_container[iZone]->GetnDV()];
+
+    for (iDV = 0; iDV  < config_container[iZone]->GetnDV(); iDV++){
+      TotalDeformation[iDV] = new su2double[config_container[iZone]->GetnDV_Value(iDV)];
+      for (iDV_Value = 0; iDV_Value < config_container[iZone]->GetnDV_Value(iDV); iDV_Value++){
+        TotalDeformation[iDV][iDV_Value] = 0.0;
+      }
+    }
     
     if (config_container[iZone]->GetDesign_Variable(0) != NO_DEFORMATION) {
       
       /*--- Definition of the Class for grid movement ---*/
       grid_movement[iZone] = new CVolumetricMovement(geometry_container[iZone], config_container[iZone]);
+
+      /*--- Save original coordinates to be reused in convexity checking procedure ---*/
+      unsigned long iPoint, nPoint = geometry_container[iZone]->GetnPoint();
+      unsigned short iDim, nDim = geometry_container[iZone]->GetnDim();
+      su2double OriginalCoordinates[nPoint][nDim];
+
+      for (iPoint = 0; iPoint < nPoint; iPoint++) {
+        for (iDim = 0; iDim < nDim; iDim++) {
+          OriginalCoordinates[iPoint][iDim] = geometry_container[iZone]->nodes->GetCoord(iPoint, iDim);
+        }
+      }      
```

Comment:
There is an easier way to do this.
`auto OriginalCoordinates = geometry_container[iZone]->nodes->GetCoord();`
or su2activematrix instead of auto.
This: `su2double OriginalCoordinates[nPoint][nDim];` is actually not standard c++